Cardigann v3 fixes

bakerboy448 opened the pull request:
#### Database Migration
NO

#### Description
- fixes and future proofs logging of rsskey/apikey from cardigann config
- fixes not using the api key in headers for searches
#### Screenshot (if UI related)

#### Todos
- Tests
- Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
-  [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #606

Qstick commented on src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannBase.cs:
```diff
@@ -292,12 +292,12 @@ protected string HandleJsonSelector(SelectorBlock selector, JToken parentObj, Di
                 }
 
-                if (setting.Type == "text" || setting.Type == "password")
+                if (setting.Type == "text" || setting.Type == "password" || setting.Name == "apikey" || setting.Name == "rsskey")
```
Don't change this one, it's not logging

Qstick commented on src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs:
```diff
@@ -1067,17 +1067,15 @@ private IEnumerable<IndexerRequest> GetRequest(Dictionary<string, object> variab
-
                 // send HTTP request
-                if (search.Headers != null)
+                if (_definition.Search.Headers != null)
```
Don't switch all these back, the search variable is set from the definition once at the beginning of the method so we don't have to call back to def field each go.

Qstick commented on src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs:
```diff
@@ -1067,17 +1067,15 @@ private IEnumerable<IndexerRequest> GetRequest(Dictionary<string, object> variab
-                    {
-                        request.HttpRequest.Headers.Add(header.Key, header.Value[0]);
-                    }
+                    var headers = ParseCustomHeaders(_definition.Search?.Headers, variables);
```
Same

Qstick reviewed (changes requested):
Shouldn't there be another change in the download method as well

https://github.com/Jackett/Jackett/pull/12504/files

bakerboy448 commented:
> Shouldn't there be another change in the download method as well
> 
> https://github.com/Jackett/Jackett/pull/12504/files

Addressed comments & pulled the download change as well - nice find!
(note: untested, but seems logical)

bakerboy448 commented on src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannBase.cs:
```diff
@@ -334,7 +334,7 @@ protected string HandleJsonSelector(SelectorBlock selector, JToken parentObj, Di
                 }
 
-                if (setting.Type != "password" && indexerLogging)
+                if (setting.Type != "password" && setting.Name != "apikey" && setting.Name != "rsskey" && indexerLogging)
```
@Qstick does it make sense to not log `setting.Type == "info"` as well since that isn't really useful?

bakerboy448 replied:
```suggestion
                if (setting.Type != "password" && setting.Name != "apikey" && setting.Name != "rsskey" && setting.Type != "info" && indexerLogging)
```
